actions/guest_file_upload.py

```diff
@@ -0,0 +1,45 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from vmwarelib import guest
+from vmwarelib import inventory
+from vmwarelib.actions import BaseAction
+
+
+class InitiateFileTransferToGuest(BaseAction):
+
+    def run(self, vm_id, username, password, directory, localpath,
+            vsphere=None):
+        """
+        Upload a file to a directory inside a guest.
+
+        Args:
+        - vm_id: MOID of the Virtual Machine
+        - username: username to perform the operation
+        - password: password of that user
+        - directory: Directory name in the guest to store the file
+        - localpath: The path to the local file consisting of the
+        -            path type (pack|path), then a colon, then
+        -            a path.
+        -            examples: pack:packname/path-inside-pack/to/file
+        -                      path:/full/path/to/file
```

Comment:
I'm really not a fan of this style of handling paths. I'd rather just specify a path, because I don't think that other people will be putting their files into their own pack, but will instead use a different location. 

But maybe I'm in the minority here. I'd like to get some input from other pack contributors before merging this.

Reply:
@userlocalhost @nmaludy you have an opinion on this?

Reply:
I'm already planning on using pack relative paths by providing scripts in my own pack and making workflows in my own pack that leverage the actions and workflows in the vSphere pack.  However I can remove the functionality since you both have reservations about it.

Reply:
Or perhaps two different optional arguments to the action where one is required?  It would be nice to be able to express that in the action yaml - define a group of options for which one (and only one for this case; at least one perhaps for other cases...) must be specified.  This would eliminate the "vm_ids" vs "vm_names" logic in scripts here.